### PR TITLE
Drop support for PHP < 8.1, upgrade to PHPUnit 10

### PR DESCRIPTION
**Pull request by Slamdunk**
Blocked by https://github.com/browscap/browscap-php/pull/640

- [x] Drop support for PHP < 8.1
- [x] Always run on `ubuntu-latest`; remove it from matrix to get a cleaner GH status name
- [x] Upgrade to PHPUnit 10

Build green on my fork:
https://github.com/Slamdunk/browscap-php/actions/runs/6941992974

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,12 +29,12 @@
...
+        "php": "~8.1.0 || ~8.2.0 || ~8.3.0",
         "ext-json": "*",
-        "guzzlehttp/guzzle": "^7.5.0",
-        "league/flysystem": "^2.5.0 || ^3.12.0",
-        "matthiasmullie/scrapbook": "^1.4.9",
-        "monolog/monolog": "^2.8.0 || ^3.2.0",
+        "guzzlehttp/guzzle": "^7.8.0",
+        "league/flysystem": "^2.5.0 || ^3.21.0",
+        "matthiasmullie/scrapbook": "^1.5.1",
+        "monolog/monolog": "^2.8.0 || ^3.5.0",
```
Comment (mimmi20) [inline composer.json]: Is it possible to remove all these duplicate versions and always use the newer version? (e.g always use monolog 3)

Comment (asgrim) [reply composer.json]: I think if we remove these, we may have compatibility issues if folks aren't using newer versions of these too. That said, if older deps are being used, the 7.2.* series will still be installable.

**Comment by Slamdunk** [comment]
Hi, since https://github.com/browscap/browscap-php/pull/640 has been merged, you should as well update the list of `Required` steps in the page https://github.com/browscap/browscap-php/settings/branches

**Review by mimmi20 — APPROVED** [review]

**Comment by asgrim** [comment]
> Hi, since #640 has been merged, you should as well update the list of `Required` steps in the page https://github.com/browscap/browscap-php/settings/branches

This was because you dropped the OS from the matrix in this PR, not relevant to #640 - anyway I've updated it to work for the new job names. Not sure why I can't just ask for "green builds" and have to specify each job but hey. We can fix that later :)

**Review by asgrim — APPROVED** [review]

**Comment by Slamdunk** [comment]
Yeah sorry, wrong link, I meant after https://github.com/browscap/browscap-php/pull/641/commits/85485dcfeee6eadfc4e229d8cbfe6f696504a8df

Thank you

**Pull request merged**